pkg/server/singleprocess/service_ui_pipeline.go

```diff
@@ -65,7 +65,17 @@ func (s *Service) UI_ListPipelines(
 		}
 		var lastRunBundle *pb.UI_PipelineRunBundle
 		if pipelineLastRun != nil {
-			lastRunBundle = &pb.UI_PipelineRunBundle{PipelineRun: pipelineLastRun}
+			job, err := s.GetJob(ctx, &pb.GetJobRequest{
+				JobId: pipelineLastRun.Jobs[0].Id,
```

Comment:
TODO: What if there is no `pipelineLastRun.Jobs[0]`? Is that possible? Panic?

Reply:
[Brian suggested](https://hashicorp.slack.com/archives/C013QT1KG9W/p1678988511969899?thread_ts=1678988372.585339&cid=C013QT1KG9W) I check for the length to avoid a panic in this case, so I've made that change.